core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -314,14 +317,26 @@ public void onStream(SdkPublisher<ByteBuffer> publisher) {
                     );
                 }
             }
-            publisher.subscribe(new IndividualPartSubscriber<>(this.individualFuture, response));
+
+            CompletableFutureUtils.forwardResultTo(upstreamFuture, resultFuture);
```

Comment:
Why do we forward result here again and do it for every part?

Reply:
Hmm, we should probably figure out why onError was invoked multiple times.

Reply:
This is the flow of events:
1) Error response is returned
2) `IndividualTransformer.exceptionOccurred()` is invoked, which calls `publisherToUpstream.error(error)`
3) In `SimplePublisher.doProcessQueue()`, in the `ON_ERROR` case `failureMessage` is set
4) `IndividualPartSubscriber.onNext()` is invoked, because there is still outstanding demand. Here, `publisherToUpstream.send(byteBuffer)` is called
5) In `SimplePublisher.doProcessQueue()`, the entry is `onNext`, but because the `failureMessage` is set, `entry.resultFuture.completeExceptionally(failureMessage.get())` is invoked

- https://github.com/aws/aws-sdk-java-v2/blob/master/utils/src/main/java/software/amazon/awssdk/utils/async/SimplePublisher.java#L260

Not sure how we can prevent 4) from happening, is it necessary?

Reply:
We could always prevent sending bytes to the `publisherToUpstream` after `IndividualTransformer` received an error, would that be a solution?